build/repo.targets

```diff
@@ -32,12 +32,28 @@
 
     <TestDependsOn />
     <TestDependsOn Condition=" '$(_RunBuild)' == 'true' ">$(TestDependsOn);Compile</TestDependsOn>
-    <TestDependsOn>$(TestDependsOn);TestProjects</TestDependsOn>
+    <TestDependsOn>$(TestDependsOn);TestProjects;TestProjectTemplates</TestDependsOn>
 
     <!-- Package modification must happen before code signing. -->
     <CodeSignDependsOn>$(CodeSignDependsOn);RemoveSharedFrameworkOnlyRefsFromNuspec</CodeSignDependsOn>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <TempDir>$(ArtifactsDir)temp</TempDir>
+  </PropertyGroup>
+
+  <Target Name="TestProjectTemplates" DependsOnTargets="BuildSharedFx" >
+    <MakeDir Directories="$(TempDir)" />
+    <Unzip
```

Comment:
I'm fine with this approach as a temporary workaround while I finish up https://github.com/aspnet/AspNetCore/issues/4257. This current implementation has a few problems, however, which you should consider:

*  Bitness on OS. You've hardcoded x64 for Windows, but we also run these tests on macOS and Linux, and we have x86, arm, and arm64 variants.
* Locations - currently hard-coded to C:/Users/rybrande...that's gotta change
* `dotnet test` will cause a rebuild. Consider `dotnet test --no-build`.
* Everything in `artifacts/` is going to be scanned by signcheck in a real CI build. This could cause some test files to be flagged because they are not signed. I would recommend putting this into `$(IntermediateDir)ProjectTemplates.Tests\` instead.

Reply:
We talked in person and decided that @natemcmaster's solution is better enough that this PR should just wait for it.